File: Easy/ArmstrongNumbers.java
import java.io.*;
public class Main {
    public static String solve(String input) {
        Integer powExponent = input.length();
        Long sum = 0L;
        for(Integer i = 0; i < input.length(); i++) {
            sum += Math.round(Math.pow(input.charAt(i) - '0', powExponent));
        }

        return (sum.equals(Long.parseLong(input))) ? "True" : "False";
    }
    
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            System.out.println(solve(line));
        }
    }
}

File: Easy/ArmstrongNumbers.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <cmath>

using namespace std;

string solve(const string &input) {
    int powExponent = input.length();
    long sum = 0L;
    for(int i = 0; i < input.length(); i++) {
        sum += floor(pow(input[i] - '0', powExponent));
    }

    return (sum == (stol(input))) ? "True" : "False";
}

int main(int argc, const char * argv[]) {
    ifstream input_stream(argv[1]);
    string input;
    
    while(getline(input_stream, input)) {
        cout << solve(input) << endl;
    }
    
    return 0;
}
